openwisp_controller/config/admin.py

```diff
@@ -859,29 +859,83 @@ def create_log_entry(user, clone):
                 action_flag=ADDITION,
             )
 
+        def save_clones(view, user, queryset, organization=None):
+            errors = []
+            # validate organization
+            if organization:
+                try:
+                    validated_org = Organization.objects.get(pk=organization)
+                except (ValidationError, Organization.DoesNotExist) as e:
+                    logger.warn(
+                        'Detected tampering in clone template '
+                        f'form by user {user}: {e}'
+                    )
+                    return
+                if not user.is_superuser and not user.is_manager(organization):
+                    logger.warn(
+                        'Detected tampering in clone template '
+                        f'form by user {user}: not authorized '
+                        f'to operate on {validated_org}.'
+                    )
+                    return
+            elif organization == '':
+                validated_org = None
+                if not user.is_superuser:
+                    logger.warn(
```

Comment:
The `warn` method is deprecated, use `warning` instead. 
```suggestion
                    logger.warning(
```